spec/models/manageiq/providers/redhat/infra_provider_spec.rb

```diff
@@ -9,13 +12,44 @@
     described_class.description.should == 'Red Hat Enterprise Virtualization Manager'
   end
 
-  it "rhevm_metrics_connect_options" do
-    h = FactoryGirl.create(:ems_redhat, :hostname => "h")
-    h.rhevm_metrics_connect_options[:host].should == "h"
+  context '#rhevm_metrics_connect_options' do
+    it "rhevm_metrics_connect_options" do
+      h = FactoryGirl.create(:ems_redhat, :hostname => "h")
+      h.rhevm_metrics_connect_options[:host].should == "h"
+    end
+
+    it "rhevm_metrics_connect_options overrides" do
+      h = FactoryGirl.create(:ems_redhat, :hostname => "h")
+      h.rhevm_metrics_connect_options(:hostname => "i")[:host].should == "i"
+    end
   end
 
-  it "rhevm_metrics_connect_options overrides" do
-    h = FactoryGirl.create(:ems_redhat, :hostname => "h")
-    h.rhevm_metrics_connect_options(:hostname => "i")[:host].should == "i"
+  context '#connect' do
+    before do
+      class Foo
+        def initialize(_)
+        end
+      end
+
+      described_class.any_instance.stub(:missing_credentials? => false)
+    end
+
+    it "connect Metrics" do
+      h = FactoryGirl.create(:ems_redhat)
+      expect(OvirtMetrics).to receive(:connect).and_return(:token)
+      expect(h.connect(:service => "Metrics")).to eq(:token)
+    end
+
+    it "connect Inventory" do
+      h = FactoryGirl.create(:ems_redhat)
+      expect(Ovirt).to receive(:const_get).and_return(Foo)
```

Comment:
The fact we access it via `const_get` still feels like implementation detail to me; that's not quite the level of "we invoke the documented API of the upstream library" the we want to prove, I don't think.